### PR TITLE
chore(sfs): modify the waitfor logic of the SFS Turbo resource

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Modify the waitfor logic of the SFS Turbo resource.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccSFSTurbo_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccSFSTurbo_basic -timeout 360m -parallel 4
=== RUN   TestAccSFSTurbo_basic
=== PAUSE TestAccSFSTurbo_basic
=== CONT  TestAccSFSTurbo_basic
--- PASS: TestAccSFSTurbo_basic (626.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       626.690s
```
```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccSFSTurbo_withEpsId"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccSFSTurbo_withEpsId -timeout 360m -parallel 4
=== RUN   TestAccSFSTurbo_withEpsId
=== PAUSE TestAccSFSTurbo_withEpsId
=== CONT  TestAccSFSTurbo_withEpsId
--- PASS: TestAccSFSTurbo_withEpsId (374.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       374.532s
```
```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccSFSTurbo_hpcShareType"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccSFSTurbo_hpcShareType -timeout 360m -parallel 4
=== RUN   TestAccSFSTurbo_hpcShareType
=== PAUSE TestAccSFSTurbo_hpcShareType
=== CONT  TestAccSFSTurbo_hpcShareType
--- PASS: TestAccSFSTurbo_hpcShareType (839.64s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       839.684s
```
```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccSFSTurbo_crypt"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccSFSTurbo_crypt -timeout 360m -parallel 4
=== RUN   TestAccSFSTurbo_crypt
=== PAUSE TestAccSFSTurbo_crypt
=== CONT  TestAccSFSTurbo_crypt
--- PASS: TestAccSFSTurbo_crypt (423.97s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       424.078s
```

* [ ] Documentation updated.
* [ ] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo.go [hunk 1]
```diff
@@ -635,39 +638,47 @@ func resourceSFSTurboDelete(ctx context.Context, d *schema.ResourceData, meta in
 	return nil
 }
 
-func waitForSFSTurboStatus(sfsClient *golangsdk.ServiceClient, shareId string) resource.StateRefreshFunc {
+func waitForSFSTurboStatus(sfsClient *golangsdk.ServiceClient, shareId string, isDelete bool) resource.StateRefreshFunc {
```
Comment (deer-hang) [inline huaweicloud/services/sfs/resource_huaweicloud_sfs_turbo.go]: It is best to split this waitFor method into two scenarios and use them respectively for create and delete.

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccSFSTurbo_hpcShareType"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccSFSTurbo_hpcShareType -timeout 360m -parallel 4
=== RUN   TestAccSFSTurbo_hpcShareType
=== PAUSE TestAccSFSTurbo_hpcShareType
=== CONT  TestAccSFSTurbo_hpcShareType
--- PASS: TestAccSFSTurbo_hpcShareType (768.35s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       768.405s
```

**Pull request merged**